src/main/java/hudson/plugins/sonar/client/WsClient.java

```diff
@@ -48,17 +51,43 @@ public WsClient(HttpClient client, String serverUrl, String username, String pas
     this.password = password;
   }
 
-  public CETask getCETask(String taskId) throws Exception {
+  public static WsClient create(HttpClient client, SonarInstallation inst) {
+    String serverUrl = StringUtils.isEmpty(inst.getServerUrl()) ? SonarInstallation.DEFAULT_SERVER_URL : inst.getServerUrl();
+    if (StringUtils.isNotEmpty(inst.getServerAuthenticationToken())) {
+      return new WsClient(client, serverUrl, inst.getServerAuthenticationToken(), null);
+    } else {
+      return new WsClient(client, serverUrl, inst.getSonarLogin(), inst.getSonarPassword());
+    }
+  }
+
+  public CETask getCETask(String taskId) {
     String url = serverUrl + API_CE_TASK + taskId;
     String text = client.getHttp(url, username, password);
+    try {
+      JSONObject json = (JSONObject) JSONSerializer.toJSON(text);
+      JSONObject task = json.getJSONObject("task");
+
+      String status = task.getString("status");
+      String componentName = task.getString("componentName");
+      String componentKey = task.getString("componentKey");
+      // No analysisId if task is pending
+      String analysisId = task.optString("analysisId", null);
+      return new CETask(status, componentName, componentKey, url, analysisId);
+    } catch (JSONException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)